File: Python/singleton_pattern/contador_singleton.py
class ContadorSingleton:
    _instance = None
    _valor = 0
    
    @property
    def valor(self):
        return self._valor
    
    def __new__(self):
        if self._instance is None:
            self._instance = super(ContadorSingleton, self).__new__(self)

        return self._instance
    
    def incrementar(self):
        self._valor += 1

File: Java/com.faculdadevincit.designpatterns.singletonpattern/src/ContadorSingleton.java
public class ContadorSingleton {
    private int valor = 0;
    private static ContadorSingleton instance;

    public static ContadorSingleton getInstance() {
        if (instance == null) {
            instance = new ContadorSingleton();
        }
        
        return instance;
    }

    public int getValor() {
        return valor;
    }


    public ContadorSingleton() {
        super();
    }

    public void incrementar() {
        valor++;
    }
}
